cmd/webhook/main.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"context"
+	"os"
+
+	"github.com/openshift-pipelines/manual-approval-gate/pkg/reconciler/webhook"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/injection"
+	"knative.dev/pkg/injection/sharedmain"
+	"knative.dev/pkg/signals"
+	kwebhook "knative.dev/pkg/webhook"
+	"knative.dev/pkg/webhook/certificates"
+)
+
+func newValidationAdmissionController(name string) func(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	return func(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+		return webhook.NewAdmissionController(ctx,
+			name,
+			"/approval-validation",
+			func(ctx context.Context) context.Context {
+				return ctx
+			},
+			true,
+		)
+	}
+}
+
+func main() {
+	serviceName := os.Getenv("WEBHOOK_SERVICE_NAME")
+	if serviceName == "" {
+		serviceName = "manual-approval-webhook"
+	}
+
+	secretName := os.Getenv("WEBHOOK_SECRET_NAME")
+	if secretName == "" {
+		secretName = "manual-approval-gate-webhook-certs"
+	}
+
+	webhookName := os.Getenv("WEBHOOK_ADMISSION_CONTROLLER_NAME")
+	if webhookName == "" {
+		webhookName = "webhook.manual.approval.dev"
+	}
```

Comment:
IMO, it will be nice to have a external function to check the environment value, if not available return the default.

```
serviceName = getEnvOrDefault("WEBHOOK_SERVICE_NAME", "manual-approval-webhook")
secretName = getEnvOrDefault("WEBHOOK_SECRET_NAME", "manual-approval-gate-webhook-certs")
webhookName = getEnvOrDefault("WEBHOOK_ADMISSION_CONTROLLER_NAME", "webhook.manual.approval.dev")
```

```
func getEnvOrDefault(key, defaultValue string) string {
	value := os.Getenv(key)
	if value == "" {
		return defaultValue
	}
	return value
}
```